src/pymatgen/core/structure.py

```diff
@@ -4717,43 +4717,53 @@ def scale_lattice(self, volume: float) -> Self:
         return self
 
     def merge_sites(self, tol: float = 0.01, mode: Literal["sum", "delete", "average"] = "sum") -> Self:
-        """Merges sites (adding occupancies) within tol of each other.
-        Removes site properties.
+        """Merges sites (by adding occupancies) within tolerance and optionally removes
+            site properties in "sum/delete" modes.
 
         Args:
             tol (float): Tolerance for distance to merge sites.
-            mode ("sum" | "delete" | "average"): "delete" means duplicate sites are
-                deleted. "sum" means the occupancies are summed for the sites.
-                "average" means that the site is deleted but the properties are averaged
-                Only first letter is considered.
+            mode ("sum" | "delete" | "average"): Only first letter is considered at this moment.
+                - "delete" means duplicate sites are deleted.
+                - "sum" means the occupancies are summed for the sites.
+                - "average" means that the site is deleted but the properties are averaged.
 
         Returns:
-            Structure: self with merged sites.
+            Structure: Structure with merged sites.
         """
-        dist_mat = self.distance_matrix
+        # TODO: change the code the allow full name after 2025-12-01
+        # TODO2: add a test for mode value, currently it only checks if first letter is "s/a"
+        if mode.lower() not in {"sum", "delete", "average"} and mode.lower()[0] in {"s", "d", "a"}:
```

Comment:
Want to hear more options on this, I guess it's beneficial to allow only full name (`"sum", "delete", "average"`) instead of checking first letter only to make use of the IDE auto-complete feature